hack/dind-cluster.sh

```diff
@@ -338,42 +245,162 @@ function nodes-are-ready() {
 EOF
   # Remove formatting before use
   template="$(echo "${template}" | tr -d '\n' | sed -e 's/} \+/}/g')"
-  local count="$("${oc}" --config="${kc}" get nodes \
-                         --template "${template}" | wc -l)"
-  test "${count}" -ge "${NODE_COUNT}"
+  local count="$("${oc}" --config="${kubeconfig}" get nodes \
+                         --template "${template}" 2> /dev/null | \
+                         grep "${node_prefix}" | wc -l )"
+  test "${count}" -ge "${expected_node_count}"
 }
 
-function wait-for-cluster() {
-  local msg="nodes to register with the master"
-  local condition="nodes-are-ready"
-  os::provision::wait-for-condition "${msg}" "${condition}"
+OS_WAIT_FOREVER=-1
+function wait-for-condition() {
+  local msg=$1
+  # condition should be a string that can be eval'd.  When eval'd, it
+  # should not output anything to stderr or stdout.
+  local condition=$2
+  local timeout=${3:-60}
+
+  local start_msg="Waiting for ${msg}"
+  local error_msg="[ERROR] Timeout waiting for ${msg}"
+
+  local counter=0
+  while ! $(${condition}); do
```

Comment:
This `while` is a little suspect -- if it's an executable string, make it so that the conditions correctly `return 0` and `return 1` when finished, and do:
```sh
while ! ${condition}; do
```

Otherwise it's not immediately obvious if you're checking the return code of the condition, or the stdout contents, or what.